### PR TITLE
DCKA 1553 - Improve wallet UI

### PR DESCRIPTION
**Pull request by Jayteekay**
## Proposed Changes
The changes included are related to the following UI fixes:
- The status bar on iOS is dark-themed in light mode, making the elements unreadable.
- Input fields with text content are not distinguishable from placeholders because the color contrast is the same.
- Icons alignment and spacing issues in Password creation pages
- Screen title font colors aren’t exactly as the design
- Toast cover back button

## Testing Instructions
To test the changes, the comparisons below can be made between the previous and new UI:

- ### Dark status bar Before -> After
 Screenshot 2023-04-04 at 12 07 41 AM -> Screenshot 2023-04-04 at 12 11 48 AM 

- ### Input color Before -> After
Screenshot 2023-04-04 at 12 15 44 AM -> Screenshot 2023-04-04 at 12 12 47 AM

- ### Password view Before -> After
Screenshot 2023-04-04 at 12 16 23 AM -> Screenshot 2023-04-04 at 12 13 48 AM

- ### Header style Before -> After 
Screenshot 2023-04-04 at 12 17 22 AM -> Screenshot 2023-04-04 at 12 14 45 AM

- ### Toast position 
Screenshot 2023-04-04 at 12 25 37 AM -> Screenshot 2023-04-04 at 12 28 44 AM

## Other Notes
⚠️ It is worth noting that a lot of the changes in this PR are repetitive and search-and-replace. Due to the refactoring of the Header component, several components and consequently several test snapshot files were changed, hence the huge PR.

### CODE REVIEW

File: src/features/verifier/CredentialVerifier.js [hunk 1]
```diff
@@ -35,21 +34,19 @@ const SCAN_OPTIONS = {
...
-            </Typography>
-            <Typography marginTop={8}>
-              {translate('credentialVerifier.description')}
-            </Typography>
-          </Box>
-        </Box>
-      </Header>
+      <Header onBack={() => navigate(Routes.APP_SETTINGS)} />
+      <Stack mx={5} mt={4}>
+        <Typography
```
Comment (maycon-mello) [inline src/features/verifier/CredentialVerifier.js:39]: Can we create a typography variant for this title? Probably it might be used in other places too

Comment (Jayteekay) [reply src/features/verifier/CredentialVerifier.js:39]: I agree. Will include this.

File: src/features/didManagement/__snapshots__/CreateNewDIDScreen.test.js.snap [hunk 2]
```diff
@@ -18,41 +18,10 @@ exports[`CreateNewDIDScreen should render correctly when form is invalid 1`] = `
...
-          </Memo(ForwardRef(Box))>
-          <Memo(ForwardRef(Box))
-            alignItems="flex-end"
-            width="80px"
-          />
-        </Box>
-      </Styled(View)>
+      <Header
+        showBackButton={true}
+        title="Create New DID"
```
Comment (mike-parkhill) [inline src/features/didManagement/__snapshots__/CreateNewDIDScreen.test.js.snap:23]: this title should be coming from the translations file - looks like we missed it originally

**Review by maycon-mello — APPROVED** [review]
looks good

**Pull request merged**